scripts/rewards-distribution/getLatestRuneAddressByAccount.ts

```diff
@@ -0,0 +1,24 @@
+import { Address } from "viem";
+import { RFoxLog, StakeLog, SetRuneAddressLog } from "./events";
+import { isLogType } from "./helpers";
+
+export const getLatestRuneAddressByAccount = (
+  orderedLogs: { log: RFoxLog; timestamp: bigint }[],
+) => {
+  const runeAddressByAccount: Record<Address, string> = {};
+
+  for (const { log } of orderedLogs) {
+    if (isLogType("Stake", log)) {
+      const stakeLog = log as StakeLog;
+      runeAddressByAccount[stakeLog.args.account] = stakeLog.args.runeAddress;
+    }
+
+    if (isLogType("SetRuneAddress", log)) {
+      const setRuneAddressLog = log as SetRuneAddressLog;
```

Comment:
If we wanted to add safety we could add `StakeLog` and `SetRuneAddressLog` type assertion functions instead of casting.

Reply:
This code is deleted in the following PR but will consider that in future implementations 🫡